lib/luminork-server/src/search/query.rs

```diff
@@ -146,6 +130,41 @@ impl SearchTerm {
     }
 }
 
+/// String methods that ignore ascii case (like eq_ignore_ascii_case for other pattern matches)
```

Comment:
I moved the methods that ignore ASCII case down into a little trait, to make it clearer that there was a correspondence between these methods and other Rust `str` methods.